internal/resources/sourcesecret/resource_sourcesecret.go

```diff
@@ -0,0 +1,403 @@
+	}
+
+	var (
+		UID  string
+		meta = common.ConstructMeta(d)
+	)
+
+	err := enableContinuousDelivery(&config, scopedFullnameData, meta)
+	if err != nil {
+		return diag.FromErr(errors.Wrapf(err, "Unable to create Tanzu Mission Control source secret entry, name : %s", sourcesecretName))
+	}
+
+	switch scopedFullnameData.Scope {
+	case commonscope.ClusterScope:
+		if scopedFullnameData.FullnameCluster != nil {
+			sourcesecretReq := &sourcesecretclustermodel.VmwareTanzuManageV1alpha1ClusterFluxcdSourceSecretRequest{
+				SourceSecret: &sourcesecretclustermodel.VmwareTanzuManageV1alpha1ClusterFluxcdSourcesecret{
+					FullName: scopedFullnameData.FullnameCluster,
+					Meta:     meta,
+					Spec:     spec.ConstructSpecForClusterScope(d),
+				},
+			}
+
+			sourcesecretResponse, err := config.TMCConnection.ClusterSourcesecretResourceService.ManageV1alpha1ClusterFluxcdSourcesecretResourceServiceCreate(sourcesecretReq)
+			if err != nil {
+				return diag.FromErr(errors.Wrapf(err, "Unable to create Tanzu Mission Control cluster source secret entry, name : %s", sourcesecretName))
+			}
+
+			UID = sourcesecretResponse.SourceSecret.Meta.UID
+		}
+	case commonscope.ClusterGroupScope:
+		if scopedFullnameData.FullnameClusterGroup != nil {
+			sourcesecretReq := &sourcesecretclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupFluxcdSourceSecretRequest{
+				SourceSecret: &sourcesecretclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupFluxcdSourcesecretSourceSecret{
+					FullName: scopedFullnameData.FullnameClusterGroup,
+					Meta:     meta,
+					Spec:     spec.ConstructSpecForClusterGroupScope(d),
+				},
+			}
+
+			sourcesecretResponse, err := config.TMCConnection.ClusterGroupSourcesecretResourceService.ManageV1alpha1ClustergroupFluxcdSourcesecretResourceServiceCreate(sourcesecretReq)
+			if err != nil {
+				return diag.FromErr(errors.Wrapf(err, "Unable to create Tanzu Mission Control cluster group sourcesecret entry, name : %s", sourcesecretName))
+			}
+
+			UID = sourcesecretResponse.SourceSecret.Meta.UID
+		}
+	case commonscope.UnknownScope:
+		return diag.Errorf("no valid scope type block found: minimum one valid scope type block is required among: %v. Please check the schema.", strings.Join(scope.CredentialTypesAllowed[:], `, `))
+	}
+
+	// always run
+	d.SetId(UID)
+
+	return resourceSourcesecretRead(ctx, d, m)
+}
+
+func resourceSourcesecretInPlaceUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	sourcesecretName, ok := d.Get(nameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read source secret name")
+	}
+
+	scopedFullnameData, _ := scope.ConstructScope(d, sourcesecretName)
+
+	if scopedFullnameData == nil {
+		return diag.Errorf("Unable to update Tanzu Mission Control source secret entry; Scope full name is empty")
+	}
+
+	_, meta, atomicSpec, err := retrieveSourcesecretUIDMetaAndSpecFromServer(config, scopedFullnameData, d)
```

Comment:
Do we need to explicitly refresh this state from server?

Reply:
done